airflow/providers/amazon/aws/operators/glue.py

```diff
@@ -0,0 +1,117 @@
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+from __future__ import unicode_literals
+
+from airflow.providers.amazon.aws.hooks.glue import AwsGlueJobHook
+from airflow.providers.amazon.aws.hooks.s3 import S3Hook
+from airflow.models import BaseOperator
+from airflow.utils.decorators import apply_defaults
+import os.path
+
+
+class AWSGlueJobOperator(BaseOperator):
+    """
+    Creates an AWS Glue Job. AWS Glue is a serverless Spark
+    ETL service for running Spark Jobs on the AWS cloud.
+    Language support: Python and Scala
+
+    :param job_name: unique job name per AWS Account
+    :type job_name: Optional[str]
+    :param script_location: location of ETL script. Must be a local or S3 path
+    :type script_location: Optional[str]
+    :param job_desc: job description details
+    :type job_desc: Optional[str]
+    :param concurrent_run_limit: The maximum number of concurrent runs allowed for a job
+    :type concurrent_run_limit: Optional[int]
+    :param script_args: etl script arguments and AWS Glue arguments
+    :type script_args: dict
+    :param connections: AWS Glue connections to be used by the job.
+    :type connections: list
+    :param retry_limit: The maximum number of times to retry this job if it fails
+    :type retry_limit:Optional[int]
+    :param num_of_dpus: Number of AWS Glue DPUs to allocate to this Job.
+    :type num_of_dpus: int
+    :param region_name: aws region name (example: us-east-1)
+    :type region_name: str
+    :param s3_bucket: S3 bucket where logs and local etl script will be uploaded
+    :type s3_bucket: Optional[str]
+    :param iam_role_name: AWS IAM Role for Glue Job Execution
+    :type iam_role_name: Optional[str]
+    """
+    template_fields = ()
+    template_ext = ()
+    ui_color = '#ededed'
+
+    @apply_defaults
+    def __init__(self,
+                 job_name='aws_glue_default_job',
+                 job_desc='AWS Glue Job with Airflow',
+                 script_location=None,
+                 concurrent_run_limit=None,
+                 script_args=None,
+                 connections=None,
+                 retry_limit=None,
+                 num_of_dpus=6,
+                 aws_conn_id='aws_default',
+                 region_name=None,
+                 s3_bucket=None,
+                 iam_role_name=None,
+                 *args, **kwargs
+                 ):
+        super(AWSGlueJobOperator, self).__init__(*args, **kwargs)
+        self.job_name = job_name
+        self.job_desc = job_desc
+        self.script_location = script_location
+        self.concurrent_run_limit = concurrent_run_limit
+        self.script_args = script_args or {}
+        self.connections = connections or []
+        self.retry_limit = retry_limit
+        self.num_of_dpus = num_of_dpus
+        self.aws_conn_id = aws_conn_id,
```

Comment:
I don't believe you meant to leave this trailing comma. That would make this a tuple.